specs/~dynamic-names.yml

```
@@ -0,0 +1,349 @@
+overview: |
+  Rationale: this special notation was introduced primarily to allow the dynamic
+  loading of partials. The main advantage that this notation offers is to allow
+  dynamic loading of partials, which is particularly useful in cases where
+  polymorphic data needs to be rendered in different ways, cases which would
+  otherwise be possible to render only with solutions that are convoluted,
+  inefficient, or both.
+  Example.
+  Let's consider the following data:
+    items: [
+      { content: 'Hello, World!' },
+      { url: 'http://example.com/foo.jpg' },
+      { content: 'Some text' },
+      { content: 'Some other text' },
+      { url: 'http://example.com/bar.jpg' },
+      { url: 'http://example.com/baz.jpg' },
+      { content: 'Last text here' }
+    ]
+  The goal is to render the different types of items in different ways: the
```

You can just end the sentence here. Otherwise you have a sentence with two colons, which is a bit uncomfortable.

```suggestion
  The goal is to render the different types of items in different ways. The
```